fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -634,7 +634,7 @@ public Void visitBetweenPredicate(BetweenPredicate node, Scope scope) {
             predicateBaseAndCheck(node);
 
             List<Type> list = node.getChildren().stream().map(Expr::getType).collect(Collectors.toList());
-            Type compatibleType = TypeManager.getCompatibleTypeForBetweenAndIn(list);
```

Comment:
`BetweenAndIn`  seems improper,  may be we can use Order,  since in, between, binary predicates based on types with Order traits: eq and lt.